src/light.rs

```diff
@@ -1,78 +1,96 @@
     }
 
     /// Enable ambient light interrupt generation.
-    pub fn enable_light_interrupts(&mut self) -> Result<(), Error<E>> {
-        self.set_flag_enable(Enable::AIEN, true)
+    pub async fn enable_light_interrupts(&mut self) -> Result<(), Error<E>> {
+        self.set_flag_enable(Enable::AIEN, true).await
     }
 
     /// Disable ambient light interrupt generation.
-    pub fn disable_light_interrupts(&mut self) -> Result<(), Error<E>> {
-        self.set_flag_enable(Enable::AIEN, false)
+    pub async fn disable_light_interrupts(&mut self) -> Result<(), Error<E>> {
+        self.set_flag_enable(Enable::AIEN, false).await
     }
 
     /// Enable clear channel ambient light saturation interrupt generation.
-    pub fn enable_light_saturation_interrupts(&mut self) -> Result<(), Error<E>> {
-        self.set_flag_config2(Config2::CPSIEN, true)
+    pub async fn enable_light_saturation_interrupts(&mut self) -> Result<(), Error<E>> {
+        self.set_flag_config2(Config2::CPSIEN, true).await
     }
 
     /// Disable clear channel ambient light saturation interrupt generation.
-    pub fn disable_light_saturation_interrupts(&mut self) -> Result<(), Error<E>> {
-        self.set_flag_config2(Config2::CPSIEN, false)
+    pub async fn disable_light_saturation_interrupts(&mut self) -> Result<(), Error<E>> {
+        self.set_flag_config2(Config2::CPSIEN, false).await
     }
 
     /// Set the clear channel ambient light interrupt low threshold.
     ///
     /// An interrupt will be generated if light interrupts are enabled and the clear data is less
     /// than this value.
-    pub fn set_light_low_threshold(&mut self, threshold: u16) -> Result<(), Error<E>> {
-        self.write_double_register(Register::AILTL, threshold)
+    pub async fn set_light_low_threshold(&mut self, threshold: u16) -> Result<(), Error<E>> {
+        self.write_double_register(Register::AILTL, threshold).await
     }
 
     /// Set the clear channel ambient light interrupt high threshold.
     ///
     /// An interrupt will be generated if light interrupts are enabled and the clear data is greater
     /// than this value.
-    pub fn set_light_high_threshold(&mut self, threshold: u16) -> Result<(), Error<E>> {
-        self.write_double_register(Register::AIHTL, threshold)
+    pub async fn set_light_high_threshold(&mut self, threshold: u16) -> Result<(), Error<E>> {
+        self.write_double_register(Register::AIHTL, threshold).await
     }
 
     /// Clear ambient light interrupt.
-    pub fn clear_light_interrupt(&mut self) -> Result<(), Error<E>> {
-        self.touch_register(Register::CICLEAR)
+    pub async fn clear_light_interrupt(&mut self) -> Result<(), Error<E>> {
+        self.touch_register(Register::CICLEAR).await
     }
 
     /// Read the color / ambient light sensor data.
     ///
-    /// Returns `nb::Error::WouldBlock` as long as the data is not ready.
+    /// Blocks as long as the data is not ready.
```

Comment:
```suggestion
    /// Returns `nb::Error::WouldBlock` as long as the data is not ready.
```
This one does not block

Reply:
Indeed. Fixed.